scaleway/data_source_image.go

```diff
@@ -94,13 +109,26 @@ func dataSourceScalewayImageRead(d *schema.ResourceData, meta interface{}) error
 		for _, v := range image.Versions {
 			for _, l := range v.LocalImages {
 				if l.Arch == d.Get("architecture").(string) && l.Zone == scaleway.Region {
-					images = append(images, l)
+					t, err := time.Parse(time.RFC3339, v.ModificationDate)
+					if err != nil {
+						log.Printf("[WARNING] could not parse modification date: %v", err.Error())
+					}
+					images = append(images, localImage{
+						MarketLocalImageDefinition: l,
+						ModificationDate:           &t,
+					})
 				}
 			}
 		}
 	}
 
-	if len(images) > 1 {
+	mostRecent := d.Get("most_recent").(bool)
+	sort.Slice(images, func(i, j int) bool {
+		return images[i].ModificationDate.Before(*images[j].ModificationDate)
+	})
+	log.Printf("[DEBUG] got %d images: %#v\n", len(images), images)
+
+	if len(images) > 1 && !mostRecent {
```

Comment:
this would not return the most recent ubuntu release, but the most recently build ubuntu release; mighy be different!
The linked issue is different tho: the names are identical, but the IDs are not. The issue of identical names with different ids can not be resolved currently.
Do you see a better option to solve this, @xunleii?

Reply:
(yep, sorry, I meant the most recent build and not the most recent release when I've said `it will return the most recent image`)

For me, it's a great option to solve this problem, but it can sometimes behave unexpectedly; as mentioned above, if we use the `most_recent` and` name_filter = "Ubuntu" `options at the same time, the resulting image will be` Ubuntu Xenial` or `Ubuntu Bionic`, or whatever, depending on the build date the more recent .

This can be easily solved by checking if in `images`, all names are the same. If so, we can select the most recent, otherwise simply return `"The query returned more than one result."`.

What do you think about this idea ? @nicolai86

Reply:
I agree that `name_filter` with `most_recent` is most likely a bad combination, but I'm not sure it deserves special handling at this point. What do you think about disallowing the combination all together right now and defering a decision until we have more data about this use case?

Reply:
I think it's a good idea

Reply:
Thank you for the good feedback!